src/sagas/workspaces.ts

```diff
@@ -96,6 +98,61 @@ export default function* workspaceSaga(): SagaIterator {
     yield* evalCode(value, context, execTime, workspaceLocation, actionTypes.EVAL_EDITOR);
   });
 
+  yield takeEvery(actionTypes.PROMPT_AUTOCOMPLETE, function*(
+    action: ReturnType<typeof actions.promptAutocomplete>
+  ) {
+    const workspaceLocation = action.payload.workspaceLocation;
+
+    context = yield select(
+      (state: IState) => (state.workspaces[workspaceLocation] as IWorkspaceState).context
+    );
+
+    const code: string = yield select((state: IState) => {
+      const prependCode = (state.workspaces[workspaceLocation] as IWorkspaceState).editorPrepend;
+      const editorCode = (state.workspaces[workspaceLocation] as IWorkspaceState).editorValue!;
+      return [prependCode, editorCode] as [string, string];
+    });
+    const [prepend, editorValue] = code;
+
+    // Deal with prepended code
+    let autocompleteCode;
+    let prependLength = 0;
+    if (!prepend) {
+      autocompleteCode = editorValue;
+    } else {
+      prependLength = prepend.split('\n').length;
+      autocompleteCode = prepend + '\n' + editorValue;
+    }
+
+    const editorNames: any = yield call(
+      getNames,
+      autocompleteCode,
+      action.payload.row + prependLength,
+      action.payload.column
+    );
+
+    const editorSuggestions = editorNames.map((name: any) => ({
+      caption: name.name,
+      value: name.name,
+      meta: name.meta,
+      score: 1000 // Prioritize suggestions from code
+    }));
+
+    const builtinSuggestions = Documentation.builtins[context.chapter] || [];
```

Comment:
Hi, was looking at the frontend and realised that the variants aren't supported yet.
![image](https://user-images.githubusercontent.com/32188705/79043359-4882e100-7c31-11ea-96ce-f8a46ad3842c.png)
I took a look and suggest something along the lines of the above. Not sure if there are any consequences of doing that right now though, and whether other variants are able to do something like this.
The inserted code in the screenshot works for Source 3 Concurrent at least.